sns/parity_math.c

```diff
@@ -500,29 +555,114 @@ static void reed_solomon_encode(struct m0_parity_math *math,
 
 #undef PARITY_MATH_REGION_ENABLE
 }
+#endif /* __KERNEL__ */
+
+#ifndef __KERNEL__
+static void isal_encode(struct m0_parity_math *math,
+			const struct m0_buf *data,
+			struct m0_buf *parity)
+{
+	uint8_t **data_frags = NULL;
+	uint8_t **parity_frags = NULL;
+	uint32_t  i;
+	uint32_t  data_count;
+	uint32_t  parity_count;
+	uint32_t  block_size;
+	int	  ret = 0;
+
+	M0_PRE(math != NULL);
+	M0_PRE(data != NULL);
+	M0_PRE(parity != NULL);
+
+	M0_ENTRY();
+
+	data_count = math->pmi_data_count;
+	parity_count = math->pmi_parity_count;
+	block_size = data[0].b_nob;
+
+	M0_ALLOC_ARR(data_frags, data_count);
```

Comment:
Since data_count does not change please move the memory allocation to m0_parity_math_init().

Reply:
Done.